commons/src/env_command.rs

```diff
@@ -0,0 +1,537 @@
+        );
+        if !self.already_streamed {
+            output.push_str(&format!(
+                "stdout: {}\nstderr{}\n",
+                self.result.stdout, self.result.stderr
+            ));
+        };
+        write!(f, "{}", output)
+    }
+}
+
+impl EnvCommand {
+    /// Main entrypoint, builds a struct with defaults and the arguments
+    /// given
+    #[allow(dead_code)]
+    pub fn new<T: IntoIterator<Item = (K, V)>, K: Into<OsString>, V: Into<OsString>>(
+        base: &str,
+        args: &[&str],
+        env: T,
+    ) -> Self {
+        EnvCommand {
+            base: base.into(),
+            args: args
+                .iter()
+                .map(std::convert::Into::into)
+                .collect::<Vec<OsString>>(),
+            env: env
+                .into_iter()
+                .map(|(k, v)| (k.into(), v.into()))
+                .collect::<HashMap<OsString, OsString>>(),
+            show_env_keys: Vec::new(),
+            on_non_zero_exit: Box::new(Err),
+        }
+    }
+
+    /// New with keys defined from the beginning
+    #[allow(dead_code)]
+    pub fn new_show_keys<T: IntoIterator<Item = (K, V)>, K: Into<OsString>, V: Into<OsString>>(
+        base: &str,
+        args: &[&str],
+        env: T,
+        show_env_keys: impl IntoIterator<Item = impl Into<OsString>>,
+    ) -> Self {
+        EnvCommand {
+            base: base.into(),
+            args: args
+                .iter()
+                .map(std::convert::Into::into)
+                .collect::<Vec<OsString>>(),
+            env: env
+                .into_iter()
+                .map(|(k, v)| (k.into(), v.into()))
+                .collect::<HashMap<OsString, OsString>>(),
+            show_env_keys: show_env_keys
+                .into_iter()
+                .map(std::convert::Into::into)
+                .collect::<Vec<OsString>>(),
+            on_non_zero_exit: Box::new(Err),
+        }
+    }
+
+    /// Attach environment variables to the struct for use when
+    /// `Display` is called on it.
+    ///
+    /// ```rust,no_run
+    /// use commons::env_command::EnvCommand;
+    /// use libcnb::Env;
+    ///
+    /// let mut env = Env::new();
+    /// env.insert("DOG", "cinco");
+    ///
+    /// let mut command =  EnvCommand::new("echo", &["hello world"], &env);
+    /// command.show_env_keys(&["DOG"]);
+    ///
+    /// assert_eq!(command.to_string(), "DOG=\"cinco\" echo \"hello world\"")
+    /// ```
+    ///
+    /// Does not change the behavior of the executing command.
+    pub fn show_env_keys(
+        &mut self,
+        keys: impl IntoIterator<Item = impl Into<OsString>>,
+    ) -> &mut Self {
+        self.show_env_keys = keys
+            .into_iter()
+            .map(std::convert::Into::into)
+            .collect::<Vec<OsString>>();
+        self
+    }
+
+    // Command is not clonable because it can contain things that are not
+    // clonable such as file descriptors. Instead we remember the
```

Comment:
```suggestion
    // Command is not cloneable because it can contain things that are not
    // cloneable such as file descriptors. Instead we remember the
```